src/main/scala/cromwell/engine/WorkflowManagerActor.scala

```diff
@@ -43,48 +45,28 @@ class WorkflowManagerActor extends Actor {
   private val workflowStore = new WriteOnceStore[WorkflowId, WorkflowActorRef]
   // This *should* be persisted
   private val workflowStates = TrieMap.empty[WorkflowId, WorkflowState]
-  private val listeners = MutableSet.empty[Promise[Unit]]
 
   def receive = LoggingReceive {
-    case SubmitWorkflow(wdl, inputs) =>
-      submitWorkflow(wdl, inputs) pipeTo sender
-
-    case WorkflowStatus(id) =>
-      sender ! workflowStates.get(id)
-
-    case NotifyCompletion(promise) =>
-      listeners += promise
-
-    case WorkflowOutputs(id) =>
-      // FIXME: What if the workflow isn't done? How best to handle?
-      workflowOutputs(id) pipeTo sender
-
-    case Shutdown =>
-      context.system.shutdown()
-
-    case WorkflowActor.Started =>
-      updateWorkflowState(sender(), WorkflowRunning)
-
-    case WorkflowActor.Done(outputs) =>
-      updateTerminalWorkflowState(WorkflowSucceeded)
-
-    case WorkflowActor.Failed(failures) =>
-      updateTerminalWorkflowState(WorkflowFailed)
-  }
-
-  private def updateTerminalWorkflowState(state: WorkflowState): Unit = {
-    listeners foreach { listener =>
-      if (!listener.isCompleted) listener.success(())
-    }
-    updateWorkflowState(sender(), state)
+    case SubmitWorkflow(wdl, inputs) => submitWorkflow(wdl, inputs) pipeTo sender
+    case WorkflowStatus(id) => sender ! workflowStates.get(id)
+    case Shutdown => context.system.shutdown()
+    // FIXME: What if the workflow isn't done? How best to handle?
+    case WorkflowOutputs(id) => workflowOutputs(id) pipeTo sender
+    case CurrentState(actor, state: WorkflowState) => updateWorkflowState(actor, state)
+    case Transition(actor, oldState, newState: WorkflowState) => updateWorkflowState(actor, newState)
+    case SubscribeToWorkflow(id) =>
+      val origSender = sender()
+      // FIXME: Silently fails if ID doesn't exist
+      workflowStore.toMap.get(id) foreach {_ ! SubscribeTransitionCallBack(origSender)}
```

Comment:
Another example to show that they're often not even consistent on the lines-per-case in the same file, yet pretty consistent on the no-newline thing:

https://github.com/akka/akka/blob/master/akka-samples/akka-sample-osgi-dining-hakkers/core/src/main/scala/akka/sample/osgi/internal/Hakker.scala

The FSM example I know had a newline between cases but that's the only one I remember offhand.

As I mentioned in the other PR I'm willing ot handwave and say if the hakkers aren't 100% consistent on the multi-line issue that we don't need to be. If you want 100% consistency I'll point to the hakkers predominantly doing things this way.